BazelExtensions/source_output_file_map_aspect.bzl

```diff
@@ -0,0 +1,78 @@
+"""
+Aspect to map source to output files (i.e. .o files generated during compilation).
+
+The mappings are generated as JSON files per target.
+"""
+
+SourceOutputFileMapInfo = provider(
+    doc = "...",
+    fields = {
+        "mapping": "Dictionary where keys are source file paths and values are at the respective .o file under bazel-out",
+    },
+)
+
+def _source_output_file_map_aspect_impl(target, ctx):
+    source_output_file_map = ctx.actions.declare_file("{}_source_output_file_map.json".format(target.label.name))
+
+    mapping = {}
+    objc_srcs = []
+    objc_objects = []
+
+    # List of source files to be mapped to output files
+    if hasattr(ctx.rule.attr, "srcs"):
+        objc_srcs = [
+            f
+            for source_file in ctx.rule.attr.srcs
+            for f in source_file.files.to_list()
+            # Handling objc only for now
+            if f.path.endswith(".m")
+            # TODO: handle swift
```

Comment:
➕ you'll need to handle how records work with `strategy=SwiftCompile=WORKER` - which will implicate assumptions on the rules here or there